src/Symfony/Installer/Manager/ComposerManager.php

```diff
@@ -0,0 +1,202 @@
+        $this->fs = new Filesystem();
+    }
+
+    public function initializeProjectConfig()
+    {
+        $composerConfig = $this->getProjectConfig();
+
+        if (isset($composerConfig['config']['platform']['php'])) {
+            unset($composerConfig['config']['platform']['php']);
+
+            if (empty($composerConfig['config']['platform'])) {
+                unset($composerConfig['config']['platform']);
+            }
+
+            if (empty($composerConfig['config'])) {
+                unset($composerConfig['config']);
+            }
+        }
+
+        $this->saveProjectConfig($composerConfig);
+    }
+
+    public function updateProjectConfig(array $newConfig)
+    {
+        $oldConfig = $this->getProjectConfig();
+        $projectConfig = array_replace_recursive($oldConfig, $newConfig);
+
+        // remove null values from project's config
+        $projectConfig = array_filter($projectConfig, function($value) { return !is_null($value); });
+
+        $this->saveProjectConfig($projectConfig);
+    }
+
+    public function getPackageVersion($packageName)
+    {
+        $composerLockFileContents = json_decode(file_get_contents($this->projectDir.'/composer.lock'), true);
+
+        foreach ($composerLockFileContents['packages'] as $packageConfig) {
+            if ($packageName === $packageConfig['name']) {
+                return $packageConfig['version'];
+            }
+        }
+    }
+
+    /**
+     * Generates a good Composer project name based on the application name
+     * and on the user name.
+     *
+     * @param $projectName
+     *
+     * @return string The generated Composer package name
+     */
+    public function createPackageName($projectName)
+    {
+        if (!empty($_SERVER['USERNAME'])) {
+            $packageName = $_SERVER['USERNAME'].'/'.$projectName;
+        } elseif (true === extension_loaded('posix') && $user = posix_getpwuid(posix_getuid())) {
+            $packageName = $user['name'].'/'.$projectName;
+        } elseif (get_current_user()) {
+            $packageName = get_current_user().'/'.$projectName;
+        } else {
+            // package names must be in the format foo/bar
+            $packageName = $projectName.'/'.$projectName;
+        }
+
+        return $this->fixPackageName($packageName);
+    }
+
+    /**
+     * It returns the project's Composer config as a PHP array.
+     *
+     * @return $this|array
+     */
+    private function getProjectConfig()
+    {
+        $composerJsonPath = $this->projectDir.'/composer.json';
+
+        if (!is_writable($composerJsonPath)) {
+//            if ($this->output->isVerbose()) {
+//                $this->output->writeln(sprintf(
+//                    " <comment>[WARNING]</comment> Project's Composer config cannot be updated because\n".
+//                    " the <comment>%s</comment> file is not writable.\n",
+//                    $composerJsonPath
+//                ));
+//            }
+
+            return $this;
```

Comment:
Yes, the `return $this;` was a leftover of the previous code.

In any case, we can remove the old warning message because returning an empty array when the Composer config is not readable will make the application work as expected.